File: app/src/main/java/com/lumiyaviewer/lumiya/ui/myava/MyAvatarActivity.java
package com.lumiyaviewer.lumiya.ui.myava;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import com.lumiyaviewer.lumiya.slproto.users.ChatterID;
import com.lumiyaviewer.lumiya.ui.common.ActivityUtils;
import com.lumiyaviewer.lumiya.ui.common.FragmentActivityFactory;
import com.lumiyaviewer.lumiya.ui.common.MasterDetailsActivity;
import com.lumiyaviewer.lumiya.ui.outfits.OutfitsFragment;
import java.util.UUID;

public class MyAvatarActivity extends MasterDetailsActivity {
    private final FragmentActivityFactory detailsFragmentFactory = new FragmentActivityFactory() {
        @Override
        public Intent createIntent(Context context, Bundle bundle) {
            return null;
        }

        @Override
        public Class<? extends Fragment> getFragmentClass() {
            return MyProfileFragment.class;
        }
    };

    @Override
    protected FragmentActivityFactory getDetailsFragmentFactory() {
        return this.detailsFragmentFactory;
    }

    @Override
    protected Bundle getNewDetailsFragmentArguments(@Nullable Bundle bundle, @Nullable Bundle bundle2) {
        UUID activeAgentID;
        return (bundle != null || (activeAgentID = ActivityUtils.getActiveAgentID(getIntent())) == null) ? super.getNewDetailsFragmentArguments(bundle, bundle2) : MyProfileFragment.makeSelection(ChatterID.getUserChatterID(activeAgentID, activeAgentID));
    }

    @Override
    protected boolean isRootDetailsFragment(Class<? extends Fragment> cls) {
        return cls == MyProfileFragment.class || cls == OutfitsFragment.class || cls == MuteListFragment.class;
    }

    @Override
    protected Fragment onCreateMasterFragment(Intent intent, @Nullable Bundle bundle) {
        return MyAvatarFragment.newInstance(ActivityUtils.getActiveAgentID(intent));
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/ui/myava/MyAvatarActivity.kt
package com.lumiyaviewer.lumiya.ui.myava

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.fragment.app.Fragment
import com.lumiyaviewer.lumiya.slproto.users.ChatterID
import com.lumiyaviewer.lumiya.ui.chat.profiles.UserProfileFragment
import com.lumiyaviewer.lumiya.ui.common.ActivityUtils
import com.lumiyaviewer.lumiya.ui.common.FragmentActivityFactory
import com.lumiyaviewer.lumiya.ui.common.MasterDetailsActivity
import com.lumiyaviewer.lumiya.ui.outfits.OutfitsFragment

class MyAvatarActivity : MasterDetailsActivity() {
    private val detailsFragmentFactory = object : FragmentActivityFactory {
        override fun createIntent(context: Context, bundle: Bundle): Intent {
            // This factory only ever embeds MyProfileFragment inside MyAvatarActivity itself
            // (see DetailsActivity.showDetails/showEmbeddedDetails); it is never launched as a
            // standalone external activity, so this path was unreachable in the original Java.
            error("MyAvatarActivity's details factory does not create standalone intents")
        }

        override fun getFragmentClass(): Class<out Fragment> = MyProfileFragment::class.java
    }

    override fun getDetailsFragmentFactory(): FragmentActivityFactory = detailsFragmentFactory

    override fun getNewDetailsFragmentArguments(bundle: Bundle?, bundle2: Bundle?): Bundle {
        if (bundle == null) {
            val activeAgentID = ActivityUtils.getActiveAgentID(intent)
            if (activeAgentID != null) {
                return UserProfileFragment.makeSelection(ChatterID.getUserChatterID(activeAgentID, activeAgentID))
            }
        }
        return super.getNewDetailsFragmentArguments(bundle, bundle2)
    }

    override fun isRootDetailsFragment(cls: Class<out Fragment>): Boolean {
        return cls == MyProfileFragment::class.java || cls == OutfitsFragment::class.java || cls == MuteListFragment::class.java
    }

    override fun onCreateMasterFragment(intent: Intent, bundle: Bundle?): Fragment {
        return MyAvatarFragment.newInstance(ActivityUtils.getActiveAgentID(intent))
    }
}
